web-server/v0.4/src/components/SessionModal/index.js

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Modal, Form, Input, Button, message, Tooltip, Icon } from 'antd';
+import axios from 'axios';
+
+const { TextArea } = Input;
+
+export default class SessionModal extends Component {
+  static propTypes = {
+    configData: PropTypes.object.isRequired,
+    styles: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      description: '',
+      visible: false,
+      confirmLoading: false,
+      generatedLink: '',
+    };
+  }
+
+  showSuccess = () => {
+    const { generatedLink } = this.state;
+
+    Modal.success({
+      title: 'Generated session link',
+      content: (
+        <div style={{ display: 'flex', flex: 1, flexDirection: 'row' }}>
+          <Input id="generatedUrl" value={generatedLink} />
+          <Button style={{ marginLeft: 8 }} icon="copy" onClick={this.copyLink}>
+            Copy Link
+          </Button>
+        </div>
+      ),
+    });
+  };
+
+  handleCancel = () => {
+    this.setState({
+      visible: false,
+    });
+  };
+
+  showModal = () => {
+    this.setState({
+      visible: true,
+    });
+  };
+
+  onGenerate = () => {
+    const { configData } = this.props;
+    const { description } = this.state;
+    const stringProp = JSON.stringify(configData);
+    axios({
+      url: 'http://localhost:4466/',
+      method: 'post',
+      data: {
+        query: `
+            mutation($config: String!, $description: String!) {
+              createUrl(data: {config: $config, description: $description}) {
+                id
+                config
+                description
+              }
+            }       
+          `,
+        variables: {
+          config: stringProp,
+          description,
+        },
+      },
+    }).then(result => {
+      this.setState({
```

Comment:
Can we handle loading via the global service you create for this query? We follow a similar pattern in other page components such as `Controllers`: https://github.com/distributed-system-analysis/pbench/blob/4b252d0848f69eae55bd4060c872b12b756c2944/web-server/v0.4/src/pages/Controllers/index.js#L17-L20